### PR TITLE
fix(web): reject path traversal during normalization

### LINKED ISSUE #1792: Security: harden file path normalization against traversal
## Description
Implement an LLM-friendly, independently shippable backlog item for **harden file path normalization against traversal**. This issue should stay narrow enough for one agent/PR while producing durable behavior, tests, and documentation where appropriate.

## Affected locations
- Input parsing and untrusted text handling
- Web/API/control-plane boundaries
- Approval, token, and state mutation paths

## Recommendation
Add the smallest safe control that addresses this risk, prefer deny-by-default behavior for unsafe paths, document operator override semantics, and include regression tests for both allowed and denied flows.

## Goals
Deliver harden file path normalization against traversal as a discrete capability or hardening change.
Make the behavior deterministic enough for autonomous workers to verify.
Keep the implementation small enough for a one-issue worker and one PR.
Capture operator/user-facing behavior in docs or help text when applicable.

## Expected outcomes
harden file path normalization against traversal is available, documented, and test-backed.
Future PM/worker handoffs can point at this issue without needing extra interpretation.
Failure modes are explicit and produce actionable messages instead of silent drift.
The change does not broaden into adjacent backlog work or unrelated refactors.

## Suggested implementation slices
Inventory the current code/docs paths that own this behavior.
Add or update the narrow model/config/schema needed for the feature or guardrail.
Implement the smallest runtime/docs change that satisfies the acceptance criteria.
Add focused fixtures/tests for success, failure, and edge cases.
Update docs/help output and handoff notes so later agents can use it.

## Acceptance criteria
- [ ] There is a dedicated implementation or documentation change for harden file path normalization against traversal.
- [ ] The control defaults to the safest practical behavior and requires explicit override for risky cases.
- [ ] At least one automated test, fixture, or deterministic verifier covers the intended behavior.
- [ ] Sensitive payloads/secrets are not printed in logs, test output, issue comments, or telemetry.
- [ ] At least one negative/edge case is covered so the failure mode is explicit.
- [ ] Docs/help text or inline operator guidance explains how to use or interpret the behavior.
- [ ] The PR description links this issue and summarizes verification evidence.
- [ ] No unrelated production behavior is changed outside the stated scope.

## LLM handoff notes
- Treat this as a one-issue/one-PR unit.
- Start by searching for existing implementations and tests before adding new surfaces.
- Prefer structured outputs and fixtures over prose-only behavior.
- If this overlaps an existing open issue, comment with evidence and stop for PM deduplication instead of creating an unscoped implementation.
- Do not perform broad refactors or change unrelated worker/PM behavior while implementing this ticket.

## Source
Created by the 2026-07-11 fourth strategic backlog expansion run requesting 100 issues. Topic: **Vulnerabilities**.

### PR DESCRIPTION
**Pull request by djm204**
## Summary
- Reject NUL bytes and parent-traversal segments in dashboard path normalization by default
- Normalize duplicate separators and dot segments while preserving explicit trusted override semantics
- Document dashboard path-field behavior and override expectations

## Test Plan
- npm --workspace @franken/web test -- src/lib/path-utils.test.ts
- npm --workspace @franken/types run build
- npm --workspace @franken/web run typecheck
- npm --workspace @franken/web run build
- npm --workspace @franken/web run lint (passes with existing warnings)

Closes #1792

### CODE REVIEW

**Comment by djm204** [comment]
@codex review

File: packages/franken-web/src/lib/path-utils.ts [hunk 1]
```diff
@@ -11,21 +11,54 @@ interface PathResult {
...
+  }
+
+  return normalized;
+}
+
+function hasParentTraversal(path: string): boolean {
+  return path.split('/').some(segment => segment === '..');
+}
+
+export function normalizePath(path: string, env: ServerEnvironment, options: NormalizePathOptions = {}): PathResult {
```
[the remark this answers is not in the document]
Comment (djm204) [reply packages/franken-web/src/lib/path-utils.ts:46]: Fixed in 330e9cfc: buildWizardLaunchConfig now normalizes repo-relative path-style fields before submission, updates the nested workflow copy, rejects unsafe absolute/traversal/NUL inputs before onLaunch can send them, and adds regression coverage for normalized and rejected launch paths.

File: packages/franken-web/src/lib/path-utils.ts [hunk 2]
```diff
@@ -11,21 +11,54 @@ interface PathResult {
...
+   */
+  allowParentTraversal?: boolean;
+}
+
+const TRAVERSAL_ERROR =
+  'Path traversal is not allowed. Use allowParentTraversal only for trusted operator-supplied paths.';
+
+function normalizeForwardSlashPath(path: string): string {
+  const absolute = path.startsWith('/');
+  const segments = path.split('/').filter(segment => segment.length > 0 && segment !== '.');
```
[the remark this answers is not in the document]
Comment (djm204) [reply packages/franken-web/src/lib/path-utils.ts:28]: Fixed in 330e9cfc: normalizePath now preserves current-directory inputs by returning "." for "." and "./", with regression coverage for both cases so directory-style fields remain non-empty.

File: packages/franken-web/src/lib/path-utils.ts [hunk 3]
```diff
@@ -11,21 +11,54 @@ interface PathResult {
...
+  const normalized = segments.join('/');
+
+  if (absolute) {
+    return `/${normalized}`;
```
[the remark this answers is not in the document]
Comment (djm204) [reply packages/franken-web/src/lib/path-utils.ts:36]: Fixed in 330e9cfc: normalizePath now detects and preserves UNC roots when normalizing Windows server paths, with a regression for \\\\server\\share\\chunks -> //server/share/chunks.

**Comment by djm204** [comment]
@codex review

**Comment by djm204** [comment]
@codex review

File: packages/franken-web/src/components/beasts/wizard-launch-config.ts [hunk 4]
```diff
@@ -118,7 +162,9 @@ export function buildWizardLaunchConfig(
     for (const prompt of selectedWorkflow.interviewPrompts) {
       const value = getPromptValue(workflow, prompt);
       if (!isBlankCatalogValue(value)) {
-        config[prompt.key] = value;
+        config[prompt.key] = typeof value === 'string' && PATH_CONFIG_KEYS.has(prompt.key)
+          ? normalizeRepoRelativeLaunchPath(prompt.key, value)
```
[the remark this answers is not in the document]
Comment (djm204) [reply packages/franken-web/src/components/beasts/wizard-launch-config.ts:178]: Fixed in d7150b67: wizard validation now rejects unsafe path-style prompts before launch, so invalid values surface as field validation instead of an uncaught launch-config error. Added validation regression coverage for design-interview outputPath and custom catalog file prompts.

File: packages/franken-web/src/components/beasts/wizard-launch-config.ts [hunk 5]
```diff
@@ -89,6 +99,37 @@ function sanitizeModuleConfig(modules: Record<string, unknown> | undefined): Rec
   return nextModules;
 }
 
+function normalizeRepoRelativeLaunchPath(fieldName: string, value: string): string {
+  if (value.startsWith('/') || value.startsWith('\\') || /^[a-zA-Z]:/.test(value)) {
+    throw new Error(`${fieldName} must be a repo-relative path without traversal.`);
```
[the remark this answers is not in the document]
Comment (djm204) [reply packages/franken-web/src/components/beasts/wizard-launch-config.ts:104]: Addressed in d7150b67 by narrowing the README/launch contract: launch submissions are repo-relative only, and absolute/drive-letter/UNC paths are rejected at the wizard boundary. WSL drive conversion remains a normalizePath behavior, but is no longer advertised for untrusted launch submissions.

File: packages/franken-web/src/components/beasts/wizard-launch-config.ts [hunk 6]
```diff
@@ -18,6 +19,15 @@ const MODULE_NUMBER_BOUNDS = {
...
 } as const;
 
+const REPO_RELATIVE_PATH_ENV: ServerEnvironment = {
+  os: 'linux',
+  platform: 'linux',
+  isWsl: false,
+  pathSeparator: '/',
+};
+
+const PATH_CONFIG_KEYS = new Set(['outputPath', 'designDocPath', 'outputDir', 'chunkDirectory']);
```
[the remark this answers is not in the document]
Comment (djm204) [reply packages/franken-web/src/components/beasts/wizard-launch-config.ts:29]: Fixed in d7150b67: catalog prompts are now normalized by path prompt semantics, including kind=file, kind=directory, and path/dir/directory key names, not only the hard-coded built-in path keys. Added regression coverage for a custom catalog file prompt.

**Comment by djm204** [comment]
@codex review
